x-pack/platform/plugins/shared/onechat/public/types.ts

```diff
@@ -52,4 +54,26 @@ export interface OnechatPluginStart {
    * Tool service contract, can be used to list or execute tools.
    */
   tools: ToolServiceStartContract;
+  /**
+   * Opens a conversation flyout.
+   *
+   * @param options - Configuration options for the flyout
+   * @returns An object containing the flyout reference and a promise that resolves when the flyout is closed
+   *
+   * @example
+   * ```tsx
+   * // Open a new conversation
+   * const { flyoutRef, promise } = plugins.onechat.openConversationFlyout({});
+   *
+   * // Programmatically close the flyout
+   * flyoutRef.close();
+   *
+   * // Wait for the flyout to close
+   * await promise;
+   * ```
+   */
+  openConversationFlyout: (options: OpenConversationFlyoutOptions) => {
+    flyoutRef: OverlayRef;
+    promise: Promise<void>;
+  };
```

Comment:
A few things:

1. NIT: no inline types for public interfaces / API contracts. Let's create a `OpenConversationFlyoutReturn` or similar instead.

2. I would have exposed a `onClose` handler in `OpenConversationFlyoutOptions` instead of exposing a promise as output. Seems more convenient to use for consumers?

3. `OverlayRef` is a little too low-level of an interface to my liking. I would introduce a `ConversationFlyoutRef`, which could just mimik the `OverlayRef` interface for now (but we could even not expose `onClose` here if it's exposed on the options instead)